src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -0,0 +1,317 @@
+    false
+  )
+  const [
+    previousAlgorithmOrderId,
+    setPreviousAlgorithmOrderId
+  ] = useState<string>()
+
+  // const { compute, isLoading, computeStepText, computeError } = useCompute()
+  // const [computeContainer, setComputeContainer] = useState(
+  //   computeOptions[0].value
+  // )
+  const isComputeButtonDisabled =
+    isJobStarting === true ||
+    file === null ||
+    computeType === '' ||
+    !ocean ||
+    !isBalanceSufficient
+  const hasDatatoken = Number(dtBalance) >= 1
+
+  async function checkPreviousOrders(ddo: DDO, serviceType: ServiceType) {
+    const orderId = await checkPreviousOrder(ocean, accountId, ddo, serviceType)
+    const assetType = ddo.findServiceByType('metadata').attributes.main.type
+    if (assetType === 'algorithm') {
+      setPreviousAlgorithmOrderId(orderId)
+      setHasPreviousAlgorithmOrder(!!orderId)
+    } else {
+      setPreviousDatasetOrderId(orderId)
+      setHasPreviousDatasetOrder(!!orderId)
+    }
+  }
+
+  async function checkAssetDTBalance(asset: DDO) {
+    const AssetDtBalance = await ocean.datatokens.balance(
+      asset.dataToken,
+      accountId
+    )
+    setHasAlgoAssetDatatoken(Number(AssetDtBalance) >= 1)
+  }
+
+  function getQuerryString(
+    trustedAlgorithmList: publisherTrustedAlgorithm[]
+  ): SearchQuery {
+    let algoQuerry = ''
+    trustedAlgorithmList.forEach((trusteAlgo) => {
+      algoQuerry += `id:"${trusteAlgo.did}" OR `
+    })
+    if (trustedAlgorithmList.length > 1) {
+      algoQuerry = algoQuerry.substring(0, algoQuerry.length - 3)
+    }
+    const algorithmQuery =
+      trustedAlgorithmList.length > 0 ? `(${algoQuerry}) AND` : ``
+    const query = {
+      page: 1,
+      query: {
+        query_string: {
+          query: `${algorithmQuery} service.attributes.main.type:algorithm AND price.type:exchange -isInPurgatory:true`
+        }
+      },
+      sort: { created: -1 }
+    }
+    return query
+  }
+
+  // must be moved to a util method used also on edit compute metadata
+  async function getAlgorithmList(): Promise<AssetSelectionAsset[]> {
+    const computeService = ddo.findServiceByType('compute')
+    let algorithmSelectionList: AssetSelectionAsset[]
+    if (
+      computeService.attributes.main.privacy.publisherTrustedAlgorithms
+        .length === 0 &&
+      !computeService.attributes.main.privacy.allowAllPublishedAlgorithms
+    ) {
+      algorithmSelectionList = []
+    } else {
+      const algorithmDDOList = await getAlgorithms(
+        getQuerryString(
+          computeService.attributes.main.privacy.publisherTrustedAlgorithms
+        ),
+        config.metadataCacheUri
+      )
+      setDdoAlgorithmList(algorithmDDOList)
+      algorithmSelectionList = await transformDDOToAssetSelection(
+        algorithmDDOList,
+        config.metadataCacheUri,
+        []
+      )
+    }
+    return algorithmSelectionList
+  }
+
+  useEffect(() => {
+    if (!ddo) return
+    async function setDatatokenSymbol(ddo: DDO) {
```

Comment:
unneeded, it's available in the DDO under `ddo.dataTokenInfo.symbol` so we do not have to fetch anything